atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -371,14 +377,23 @@ private static SpecificTableSweeper initializeSweepEndpoint(
         return specificTableSweeper;
     }
 
-    private static SweepBatchConfig getSweepBatchConfig(SweepConfig sweepConfig) {
+    private static SweepBatchConfig getSweepBatchConfig(SweepConfig sweepConfig, KeyValueServiceConfig kvsConfig) {
         return ImmutableSweepBatchConfig.builder()
                 .maxCellTsPairsToExamine(sweepConfig.readLimit())
-                .candidateBatchSize(sweepConfig.candidateBatchHint())
+                .candidateBatchSize(sweepConfig.candidateBatchHint()
+                        .orElse(getDefaultSweepCandidateBatchHint(kvsConfig)))
                 .deleteBatchSize(sweepConfig.deleteBatchHint())
                 .build();
     }
 
+    // TODO(nziebart): this is a hack until we fix cassandra's getCandidateCellsForSweep to behave like the other KVSs
+    private static int getDefaultSweepCandidateBatchHint(KeyValueServiceConfig kvsConfig) {
+        if (kvsConfig.type().equals("cassandra")) {
+            return AtlasDbConstants.DEFAULT_SWEEP_CANDIDATE_BATCH_HINT_CASSANDRA;
```

Comment:
Can we update the docs here: http://palantir.github.io/atlasdb/html/cluster_management/sweep.html?highlight=sweep#tunable-configuration-options as the default section is no longer true.

Reply:
I think the docs change was added to the other PR, but this one will merge into that so should be fine :P.

Reply:
oh whoops. let me fix that

Reply:
i'd actually rather merge these two separately to keep the revert as clean as possible in the history